### DISCUSSION TITLE
"Open in Elk" bookmarklet

### DISCUSSION THREAD

**Discussion by jgierer12** [elk-zone/elk#1898] [Show and tell]
I really love Elk, but I had a small annoyance: There's no quick way to open normal Mastodon pages in Elk. This is of course not an issue with Elk itself, it's a general problem when trying to interact with external Mastodon instances. But Elk makes it super simple to solve with a bookmarklet. I use it all the time and I thought it might be useful to other people, so I'm sharing it here.

Simply make a new bookmark, title it "Open in Elk" (or whatever you want), and set the URL to the following:

```js
javascript:(function(){ window.open('https://elk.zone/' + window.location.href, '_self'); })();
```

Now, whenever you're on any Mastodon page, such as https://mastodon.social/@tur1ng, you can click on the bookmarklet to open that page in Elk!

**Comment by patak-cat** [comment]
Nice! Maybe you could join https://chat.elk.zone and also share it there? 🧡

**Reply by jgierer12** [reply]
Thanks, I'll do that!

**Comment by superusercode** [comment]
Is it possible to update this to support lemmy/kbin as well?

**Reply by jgierer12** [reply]
I don't use those platforms so I'm not sure what you mean. I don't think Elk works as a frontend for them, does it?

**Reply by superusercode** [reply]
lemmy and kbin use activitypub, though not all instances federate properly. On new versions of mastodon (I believe 4,1.x) you can copy in a lemmy link (i.e. https://lemmy.ml/post/1915537) and click the "open URL in mastodon" button that resolves a post in your client (turning the link into https://mastodon.social/@gkd@lemmy.ml/110691380567936749). Similarly it is able to do this for kbin links (https://kbin.social/m/lemmyshitpost@lemmy.world/t/176130/Yelp-for-coworkers-would-be-hilarious-or-grind-productivity-to-a) becomes (https://mastodon.social/@idunn@lemmy.fmhy.ml/110695813262898938).

**Reply by jgierer12** [reply]
Interesting. If Mastodon has an API to do this URL conversion, it's certainly be possible to integrate that into the bookmarklet.

Alternatively you could hope/request for Elk to do this conversion internally, so for example https://elk.zone/https://lemmy.ml/post/1915537 would just work. Then my original bookmarklet would also work as expected. Not sure how willing the maintainers would be to add this though, since Elk is primarily a Mastodon client.
